onnxscript/optimizer/_constant_folding.py

```diff
@@ -71,6 +71,14 @@ def _process_constant_node(node: ir.Node) -> None:
     if attr_value is None or not isinstance(attr_value, ir.Attr):
         return
 
+    # Even if this is an attribute, the value property might not be set, which
+    # happens e.g. in case of attribute references, i.e., ref_attr_name is set
+    if attr_value.value is None:
```

Comment:
Is there a test you can help add?

Reply:
Hm, not right off the bat, stumbled upon this in the middle of some lengthy debugging. I am not even sure how to properly test this, as there is not really any observable consequence besides "not crashing"?  Maybe some test case could be constructed involving a function with both, a proper attribute embedded within (constant-foldable) and another one referencing an attribute from the outside (not constant-foldable, as long as you don't inline the function)?

Reply:
I am thinking you can create a model with a single constant node whose attribute value is empty. Run the pass on the model, and observe that it is unchanged. Would that be good enough?